src/mavlink/manager.rs

```diff
@@ -1 +1,203 @@
+        std::thread::Builder::new()
+            .name("MavReceiver".into())
+            .spawn(move || Manager::receiver_loop(connection))
+            .expect("Failed to spawn MavReceiver thread");
+
+        this
+    }
+}
+
+impl Manager {
+    // Construct our manager, should be done inside main
+    #[instrument(level = "debug")]
+    pub fn init() {
+        MANAGER.as_ref();
+    }
+
+    #[instrument(level = "debug", skip(inner))]
+    fn receiver_loop(inner: Arc<Mutex<Connection>>) {
+        loop {
+            std::thread::sleep(std::time::Duration::from_micros(10));
+
+            // Receive from the Mavlink network
+            let (header, message) = match inner.lock().unwrap().connection.recv() {
+                Ok(message) => message,
+                Err(error) => {
+                    if let mavlink::error::MessageReadError::Io(io_error) = &error {
+                        if io_error.kind() == std::io::ErrorKind::WouldBlock {
+                            trace!("Blocked I/O, probably disconnected!");
+                            continue;
+                        }
+                    }
+                    error!("Failed receiving message from Mavlink Connection: {error:?}");
+                    break;
+                }
+            };
+
+            trace!("Message received: {header:?}, {message:?}");
+
+            // Send the received message to the cameras
+            if let Err(error) = inner
+                .lock()
+                .unwrap()
+                .sender
+                .send(Message::Received((header, message)))
+            {
+                error!("Failed handling message: {error:?}");
+                continue;
+            }
+        }
+    }
+
+    #[instrument(level = "debug", skip(inner))]
+    fn sender_loop(inner: Arc<Mutex<Connection>>) {
+        let mut receiver = { inner.lock().unwrap().sender.subscribe() };
+
+        loop {
+            loop {
+                std::thread::sleep(std::time::Duration::from_micros(10));
+
+                // Receive answer from the cameras
+                let (header, message) = match receiver.try_recv() {
+                    Ok(Message::ToBeSent(message)) => message,
+                    Err(broadcast::error::TryRecvError::Closed) => {
+                        unreachable!(
+                            "Closed channel: This should never happen, this channel is static!"
+                        );
+                    }
+                    Ok(Message::Received(_)) | Err(_) => continue,
+                };
+
+                // Send the response from the cameras to the Mavlink network
+                if let Err(error) = inner.lock().unwrap().connection.send(&header, &message) {
+                    error!("Failed sending message to Mavlink Connection: {error:?}");
+                    break;
+                }
+
+                trace!("Message sent: {header:?}, {message:?}");
+            }
+
+            // Reconnects
+            {
```

Comment:
The outside loop does not have a sleep, and that's super dangerous.
```suggestion
            {
                std::thread::sleep(std::time::Duration::from_millis(500));
```

Reply:
Placed right before the end of the context